ls1intum/Artemis
# `Quiz exercises`: Handle partial points in statistics

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


The quiz point statistics currently cannot handle partial points of a quiz (i.e. with decimal places). They where not visible at all. This is fixed by this PR. 
Additionally, I fixed a few minor things:

- The average in the quiz statistics (second chart) is now rounded to 2 decimal places in order to prevent too long numbers here.
- I removed the rotation of 45 degree of the datalabels
- The labels are now centered over the corresponding bars

### Description

I adapted the rounding of the results on server side. So far, this was done depending on the course configuration. This is not applicable here, as we distribute the scores in discrete (integer) buckets. Therefore, we now round the score to the nearest natural number in order to make it displayable in the chart.
I additionally adapted the labels on the x axis in order to make the rounding more transparent to the user.

### Steps for Testing

Prerequisites:
- 1 Instructor
- (at least) 1 Student

1. Log in to Artemis as Instructor
2. Create a quiz exercise where it is possible to achieve non-integer points
3. Participate with students in the exercise. Try to achieve partial points.
4. As Instructor, open the quiz statistics. The point statistic is the first chart that is presented to you
5. Compare the individual results displayed to the students with the point distribution of the quiz. The statstic should round the student points and display it respectively in the chart.
6. Check the quiz statistic. Make sure that the average is rounded to one decimal place (if not zero).

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage






### Screenshots


The problem with the average that occurred (and should be fixed with this PR):

![image (4)](https://user-images.githubusercontent.com/80622272/148995237-fcd3c171-f8d5-4efe-8e6b-0bf6aa31eff7.png)

Currently, the quiz point statistic is presenting the distribution as if only integers were possible as scores:
![image](https://user-images.githubusercontent.com/80622272/149381116-f41e19a4-1762-4f9f-ad5d-07eb471fb355.png)

With this PR, the x axis labels are adapted in order to display the score range summarized in each bar:
![image](https://user-images.githubusercontent.com/80622272/149381309-db751bb3-56b9-4e39-86aa-400551068c4a.png)
